Replace existing ClickOutside class component imports

amanda-anderson opened the pull request:
# Changelog
## Fixed
- Fixed issue where you were unable to close the DatePicker calendar once opened.

amanda-anderson committed:
Convert ClickOutside to functional component

amanda-anderson committed:
Code cleanup

amanda-anderson committed:
Code cleanup

luucvanderzee commented on front/app/utils/containers/clickOutside/index.tsx:
```diff
@@ -16,115 +18,100 @@ type Props = {
+}: Props) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
```
I think this `useEffect` needs to be reworked a bit. Now, if `closeOnClickOutsideEnabled` is `true`, we are creating this `handle` function for nothing, because it never gets added as event listener due to the check on line 66. On top of that, in this case where we don't end up adding this `handle` function as an event listener, we still try to clean it up in `removeEventListeners`. 

These things probably won't cause any problems functionally, but I think it would be cleaner to short-circuit immediately at the start of the `useEffect` if `closeOnClickOutsideEnabled` is `true`.

luucvanderzee replied:
Also, the name `closeOnClickOutsideEnabled` is a bit awkward. Maybe just `onClickOutsideEnabled`? Or, instead of a separate prop, just make the `onClickOutside` prop optional, and when it's not passed we disable the behavior?

amanda-anderson replied:
Good point! I'll re-work the code a bit :relaxed::+1:

luucvanderzee commented on front/app/utils/containers/clickOutside/index.tsx:
```diff
@@ -16,115 +18,100 @@ type Props = {
-  handleRef = (element: HTMLDivElement) => {
-    this.container = element;
-    this.props.setRef && this.props.setRef(element);
+  const handleOnMouseEnter = (event: MouseEvent) => {
```
I think these handle functions are not really necessary- you can just pass `onMouseEnter` directly to `onMouseEnter` of the `Box`. Same for the other two handle functions below

amanda-anderson replied:
Good point! I'll remove them :+1:

amanda-anderson committed:
Use new component library component for click outside

amanda-anderson commented:
@luucvanderzee I actually just realized we had two ClickOutside components, and the one in the component library had already been converted into a functional component! :) 

So I've just replaced the import statements instead, and deleted the unnecessary ClickOutside component :+1:

luucvanderzee reviewed (approved):
Cool! The only thing I would change is to use the standard `@citizenlab/component-library` import, and explicitly export this `clickOutside` from the`index` of the library. Just because it seems to be the convention

amanda-anderson committed:
Use correct component library imports

amanda-anderson committed:
Merge branch 'master' into convert-click-outside-to-functional

amanda-anderson commented:
E2E test failing is a flaky one already on master, so I'll go ahead and merge.
